control-plane/csi-driver/src/bin/controller/main.rs

```diff
@@ -1,24 +1,61 @@
-use tracing::info;
+use config::CsiControllerConfig;
+use csi_driver::client::{RestApiClient, REST_CLIENT};
+use stor_port::types::v0::openapi::clients;
 
+use anyhow::anyhow;
 use clap::{Arg, ArgMatches};
-mod client;
+use tracing::info;
+
 mod config;
 mod controller;
 mod identity;
 mod pvwatcher;
 mod server;
 
-use client::{ApiClientError, CreateVolumeTopology, IoEngineApiClient};
-use config::CsiControllerConfig;
-
 const CSI_SOCKET: &str = "/var/tmp/csi.sock";
 const CONCURRENCY_LIMIT: usize = 10;
 const REST_TIMEOUT: &str = "30s";
 
+/// Initialize API client instance. Must be called prior to
+/// obtaining the client instance.
+pub(crate) fn initialize_rest_api() -> anyhow::Result<()> {
+    if REST_CLIENT.get().is_some() {
+        return Err(anyhow!("API client already initialized"));
+    }
+
+    let cfg = CsiControllerConfig::get_config();
+    let endpoint = cfg.rest_endpoint();
+
+    let url = clients::tower::Url::parse(endpoint)
+        .map_err(|error| anyhow!("Invalid API endpoint URL {}: {:?}", endpoint, error))?;
+    let concurrency_limit = cfg.create_volume_limit() * 2;
+    let tower = clients::tower::Configuration::builder()
+        .with_timeout(cfg.io_timeout())
+        .with_concurrency_limit(Some(concurrency_limit))
+        .build_url(url)
+        .map_err(|error| {
+            anyhow::anyhow!(
+                "Failed to create openapi configuration, Error: '{:?}'",
+                error
+            )
+        })?;
+
+    REST_CLIENT.get_or_init(|| RestApiClient {
```

Comment:
can we not use the oncecell now? or using it without self?

Reply:
I did not want to change that bit of code, as it's add lot of unrelated to changes in this PR.

Reply:
Do want me to add this in this PR?

Reply:
We can address this later then